iotilecore/iotile/core/hw/debug/firmware_image_analyzer.py

```diff
@@ -0,0 +1,150 @@
+from iotile.core.utilities.intelhex import IntelHex
+from iotile.core.exceptions import *
+from tempfile import NamedTemporaryFile
+import os
+import subprocess
+import struct
+
+class FirmwareImageAnalyzer:
+    """A firmware image analyzer using IntelHex"""
+
+    KNOWN_FIRMWARE_NAMES = ['boot52', 'NRF52 ', 'firmPP', 'UART  ', 'firmFF']
```

Comment:
I wouldn't hardcode these names here.  It just means that we will have to update this package whenever we add another name and as I see the intended use-case, it's just a secondary check that we've found the info block.  I don't think that secondary check is worth the hassle of needing to keep this list in check with some internal list of firmware names.

Reply:
This is true. However, I ran into the case where I saw the magic number elsewhere in the firmware where it was *not* the cdb app info block. That is why I added this secondary check. I'm not too sure what other parameters I can check within the `CDBApplicationInfoBlock`. Maybe I can just attempt to `.decode("utf-8")` to make sure that the `name` is in fact a string?

Reply:
That's one potential option.  An alternative is to validate the checksum embedded in the application block against the firmware image itself.  This second option though won't work for legacy controller firmwares but will work for all tile firmware.

I would probably recommend just verifying that the name is a utf-8 encoded string and that should be good enough.

Reply:
Got it, fixed in next push.